dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,23 @@
 package dfa // leave this line in the file
 
-// TODO: replace this comment with your implementation
+case class State(label: String)
+
+case class Transition(from: State, to: State, symbol: Char)
+
+class DFA(val states: Set[State], val transitions: Set[Transition], val start: State, val accept: Set[State]):
```

Comment:
I was not able to get this portion working, but it is interesting to see a full implementation. The function accepts to run the given input is cool! The filter is a neat implementation.